File: Python/Part-1/PalindromeNumber.py
"""
Problem: Check whether a given number is a palindrome.
Example:
    Input  : 121
    Output : Palindrome Number
Logic:
    - Reverse the number and compare with original
"""

num = int(input("Enter number: "))
original_num = num
rev = 0
temp = num

while temp > 0:
    rem = temp % 10
    rev = rev * 10 + rem
    temp = temp // 10

if original_num == rev:
    print(f"Given number {original_num} is a Palindrome number")
else:
    print(f"Given number {original_num} is NOT a Palindrome number")


"""
Output:-
--------
Enter number: 18581
Given number 18581 is a Palindrome number
"""

File: Java/Part-1/PalindromeNumber.java
/*

Problem:Write a Java program to check whether a given number is a palindrome or not.
------------------------------------------------------------------------------------
Example:
    Input  : 121
    Reverse: 121
    Output : Palindrome Number

What is a Palindrome?
    A number that reads the same forward and backward.

Logic Used:
    - Store the original number
    - Reverse the number using:
          rem = num % 10
          rev = rev * 10 + rem
          num = num / 10
    - Compare reversed number with original number

--------------------------------------------------------------------------------------
*/

import java.util.*;
class PalindromeNumber
{
    public static void main(String[] args)
    {
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter number:");

        int num = sc.nextInt();
        int original_num = num;
        int rev = 0;

        while(num > 0)
        {
            int rem = num % 10;
            rev = rev * 10 + rem;
            num = num / 10;
        }

        if(original_num == rev)
        {
            System.out.println("Given number " + original_num + " is a Palindrome number");
        }
        else
        {
            System.out.println("Given number " + original_num + " is NOT a Palindrome number");
        }
    }
}

/*
Output:-
-------
Enter number:
121
Given number 121 is a Palindrome number
*/